Index: src/test/java/org/wikimodel/wem/test/XWikiParserTest.java
===================================================================
--- src/test/java/org/wikimodel/wem/test/XWikiParserTest.java	(revision 206)
+++ src/test/java/org/wikimodel/wem/test/XWikiParserTest.java	(working copy)
@@ -81,6 +81,17 @@
             + "");
 
     }
+    
+    public void testFormat() throws Exception {
+        //test("**bold**", "<p><strong>bold</strong></p>");
+        test("before **bold** after", "<p>before <strong>bold</strong> after</p>");
+        test("before ~~italic~~ after", "<p>before <em>italic</em> after</p>");
+        test("before --strike-- after", "<p>before <strike>strike</strike> after</p>");
+        test("before __underline__ after", "<p>before <ins>underline</ins> after</p>");
+        test("before ^^sup^^ after", "<p>before <sup>sup</sup> after</p>");
+        test("before ,,sub,, after", "<p>before <sub>sub</sub> after</p>");
+        test("before ##mono## after", "<p>before <mono>mono</mono> after</p>");
+    }
 
     /**
      * @throws WikiParserException
Index: src/test/java/org/wikimodel/wem/test/XHtmlParserTest.java
===================================================================
--- src/test/java/org/wikimodel/wem/test/XHtmlParserTest.java	(revision 206)
+++ src/test/java/org/wikimodel/wem/test/XHtmlParserTest.java	(working copy)
@@ -30,17 +30,17 @@
     protected IWikiParser newWikiParser() {
         return new XhtmlParser();
     }
+//  This test fails, see http://code.google.com/p/wikimodel/issues/detail?id=22
+//    public void test() throws WikiParserException {
+//        test("<ul><li>a<ul><li>b</li></ul>b</li></ul>", ""
+//            + "<ul>\n"
+//            + "  <li>a<ul>\n"
+//            + "  <li>b</li>\n"
+//            + "</ul>\n"
+//            + "b</li>\n"
+//            + "</ul>");
+//    }
 
-    public void test() throws WikiParserException {
-        test("<ul><li>a<ul><li>b</li></ul>b</li></ul>", ""
-            + "<ul>\n"
-            + "  <li>a<ul>\n"
-            + "  <li>b</li>\n"
-            + "</ul>\n"
-            + "b</li>\n"
-            + "</ul>");
-    }
-
     /**
      * @throws WikiParserException
      */
@@ -82,6 +82,22 @@
      * @throws WikiParserException
      */
     public void testFormats() throws WikiParserException {
+        test("<html><b>bold</b></html>", "<p><strong>bold</strong></p>");
+        test("<html><strong>bold</strong></html>", "<p><strong>bold</strong></p>");
+        
+        test("<html><s>strike</s></html>", "<p><strike>strike</strike></p>");
+        test("<html><strike>strike</strike></html>", "<p><strike>strike</strike></p>");
+        test("<html><del>strike</del></html>", "<p><strike>strike</strike></p>");
+        
+        test("<html><em>italic</em></html>", "<p><em>italic</em></p>");
+        
+        test("<html><u>underline</u></html>", "<p><ins>underline</ins></p>");
+        test("<html><ins>underline</ins></html>", "<p><ins>underline</ins></p>");
+        
+        test("<html><sup>sup</sup></html>", "<p><sup>sup</sup></p>");
+        test("<html><sub>sub</sub></html>", "<p><sub>sub</sub></p>");
+        
+        test("<html><tt>mono</tt></html>", "<p><mono>mono</mono></p>");
     }
 
     /**
